### ISSUE TITLE
Feature Request: enable lua functions to be used as commands

### ISSUE THREAD

**Issue by Corey-Keller** [lazytanuki/nvim-mapper#12]
It's the biggest reason I'm using folke/which-key.nvim.

**Comment by issadarkthing** [comment]
You can bind lua function to a keybinding like this.
```lua
-- lua/plugin/telescope.lua
-- fallback to find_files if git_files can't find a .git directory
local M = {}

M.project_files = function()
  local opts = {} -- define here if you want to define something
  local ok = pcall(require'telescope.builtin'.git_files, opts)
  if not ok then require'telescope.builtin'.find_files(opts) end
end

return M
```
And in the `mappings.lua` file.
```lua
local Mapper = require('nvim-mapper')

Mapper.map("n", "<leader>f", "<cmd>lua require'plugin/telescope'.project_files()<cr>", "Telescope", "telescope_find", "Find files in git project and fallback to files if .git directory not present")
```

**Comment by Corey-Keller** [comment]
@lazytanuki that technically works, but has nowhere near the convenience of being able to use the function directly:
```
Mapper.map("n", "<leader>f", function()
  local opts = {} -- define here if you want to define something
  local ok = pcall(require'telescope.builtin'.git_files, opts)
  if not ok then require'telescope.builtin'.find_files(opts) end
end,
"Telescope", "telescope_find",
"Find files in git project and fallback to files if .git directory not present")
```

Making lua functions work as a proper command makes it easier, quicker, and simpler to utilize the more complex mappings that power users would often want to use.

**Comment by lazytanuki** [comment]
Hi !
This plugin only forwards some of its arguments to the native `vim.api.nvim_set_keymap` functions.
This would be out of scope of this plugin I guess, maybe you it would be a feature request for the neovim project ?

**Comment by Corey-Keller** [comment]
That's unfortunate, as I feel like that functionality would increase the versatility of nvim-mapper.

It's your plugin though, so you do you :)

**Comment by lucax88x** [comment]
Probably it's using [astronauta](https://github.com/tjdevries/astronauta.nvim) which contains this [PR](https://github.com/neovim/neovim/pull/13823) that allows functions.

I guess.

**Comment by lazytanuki** [comment]
Merged in #22
